src/network/localconnector.cpp

```diff
@@ -0,0 +1,380 @@
+                         });
+    if (connectedServerAttributesPtr != m_connectedTcpServersAttrib.end())
+    {
+        return;
+    }
+
+    auto serverAttributesPtr =
+            std::find_if(m_discoveredTcpServersAttrib.begin(),
+                         m_discoveredTcpServersAttrib.end(),
+                         [userName] (const ServerData & serverData)
+                         {
+                             return serverData.m_name == userName;
+                         });
+    // Request sharing from user with a given userName
+    if (serverAttributesPtr != m_discoveredTcpServersAttrib.end())
+    {
+        bool isAllowedToStartSharing = true;
+        if (isSelfInitiated)
+        {
+            Message message;
+            message.m_type       = MessageType::MessageTypeSharingControlMessage;
+            message.m_sourceName = m_tcpService->getServerAttributes().m_name;
+            message.m_data       = QString("start");
+            isAllowedToStartSharing =
+                    m_tcpService->sendDataToTcpServer(message.toJsonQString(),
+                                                      *serverAttributesPtr);
+        }
+        if (isAllowedToStartSharing)
+        {
+            m_connectedTcpServersAttrib.push_back(*serverAttributesPtr);
+
+            qDebug() << "sharing was started";
+            emit connectedUsersUpdated(getConnectedUsers());
+        }
+    }
+}
+// ==========================================================================================
+// ==========================================================================================
+// ==========================================================================================
+//                                                                               STOP SHARING
+void DefaultLocalConnector::stopSharing(const QString userName,
+                                        const bool isSelfInitiated)
+{
+    auto serverAttributesPtr =
+            std::find_if(m_connectedTcpServersAttrib.begin(),
+                         m_connectedTcpServersAttrib.end(),
+                         [userName] (const ServerData & serverData)
+                         {
+                             return serverData.m_name == userName;
+                         });
+
+    // Request sharing from user with a given userName
+    if (serverAttributesPtr != m_connectedTcpServersAttrib.end())
+    {
+        if (isSelfInitiated)
+        {
+            Message message;
+            message.m_type       = MessageType::MessageTypeSharingControlMessage;
+            message.m_sourceName = m_tcpService->getServerAttributes().m_name;
+            message.m_data       = QString("stop");
+            m_tcpService->sendDataToTcpServer(message.toJsonQString(),
+                                              *serverAttributesPtr);
+        }
+
+        m_connectedTcpServersAttrib.erase(serverAttributesPtr, std::next(serverAttributesPtr));
+
```

Comment:
It is enough to pass 1 argument:
m_connectedTcpServersAttrib.erase(serverAttributesPtr);

Reply:
Repaired now.